### ISSUE TITLE
DependentInput causes errors and a blank page in 'next'

### ISSUE THREAD

**Issue by afilp** [marmelab/admin-on-rest#1607]
**What you were expecting:**
`<DependentInput>` to show the child when a value is shown

**What happened instead:**
The page crashes (blank page), with many errors:

![image](https://user-images.githubusercontent.com/7850073/36940310-3cb8e5d2-1f49-11e8-9c5d-81386e3b8f27.png)

![image](https://user-images.githubusercontent.com/7850073/36940312-4bf72752-1f49-11e8-9325-cb03d23b0913.png)

...and more errors.

**Related code:**
````
      <DependentInput label="SUBCATEGORY" source="subCat_id" reference="subCategories" allowEmpty
                      dependsOn="cat_id"
                      perPage={100}
                      sort={{field: 'name', order: 'ASC'}}
      >
        <SelectInput optionText="name"
                  options={{
                    fullWidth: true,
                  }}
        />
      </DependentInput>
````
* Admin-on-rest version:  latest 'next' commit (but same happens with `^2.0.0-beta3`

**Comment by afilp** [comment]
I did a lot of debugging and realized that this change seems to make it work (for 'next' branch):

![image](https://user-images.githubusercontent.com/7850073/36940976-34cb0ed0-1f59-11e8-8860-c4d07aaea148.png)

I did not submit a PR because most likely I am missing something and the solution should be better (plus more code needs to be changed, specs, etc.)

Can you please comment if this is the proper solution?

**Comment by fzaninotto** [comment]
Fixed by #1688
